controller/api/destination/watcher/k8s.go

```diff
@@ -50,25 +50,41 @@ func getHostAndPort(authority string) (string, Port, error) {
 // GetServiceAndPort is a utility function that destructures an authority into
 // a service and port.  If the authority does not represent a Kubernetes
 // service, an error is returned.  If no port is specified in the authority,
-// the HTTP default (80) is returned as the port number.
-func GetServiceAndPort(authority string) (ServiceID, Port, error) {
+// the HTTP default (80) is returned as the port number.  If the authority
+// is a pod DNS name then the pod hostname is also returned as the 3rd return
+// value.  See https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/.
+func GetServiceAndPort(authority string, clusterDomain string) (ServiceID, Port, string, error) {
```

Comment:
`GetServiceAndPort` now returns a `ServiceID`, `Port`, `string` and `error`.

---

It looks inconsistent that the pod hostname does not have a type while for example port does 

---

In addition returning a hostname from a `GetServiceAndPort` function seems confusing. IMO it's time to rename the function. Maybe `ParseAuthority`? Similar to go's `url.Parse`?

Reply:
Great suggestion, thanks.